Implements dropdown for filter selection

anishaswain opened the pull request:
https://user-images.githubusercontent.com/16955978/117258507-adc3f880-ae6a-11eb-8131-1869609bfe84.mov

gurbirkalsi commented on src/components/TableFilterSelection/index.js:
```diff
@@ -1,10 +1,18 @@
-
-import Button from '../Button';
+import styles from './index.less';
+// import Button from '../Button';
```
Do we need this comment here?

anishaswain replied:
Removed

gurbirkalsi reviewed (changes requested):
Great start @anishaswain! A few design considerations & bugs we'll need to address before merging: 
- Hovering over the toggled filter selection component highlights every field name in red
<img width="842" alt="Screen Shot 2021-05-06 at 4 59 46 PM" src="https://user-images.githubusercontent.com/4149652/117366038-3011e400-ae8e-11eb-86a1-93d21ff40ca3.png">

- Closing a selected filter from a field's dropdown menu doesn't remove the selected filter
- The width of the dropdown menu for a field scales according to the page width. Can we consider setting `width: auto` to set the width for the menu? 

<img width="654" alt="Screen Shot 2021-05-06 at 5 10 14 PM" src="https://user-images.githubusercontent.com/4149652/117365865-ede8a280-ae8d-11eb-98ab-fdab954fc825.png">

- Clicking the "Save" button doesn't close the toggled filter selection component
- We can utilize the [Toolbar](https://www.patternfly.org/v4/components/toolbar) component to align the "Apply Filters" and "Clear Filters" buttons with "Compare Iterations"

<img width="744" alt="Screen Shot 2021-05-06 at 5 09 21 PM" src="https://user-images.githubusercontent.com/4149652/117365784-ce517a00-ae8d-11eb-9e85-c1bb2adf7d3a.png">

anishaswain commented:
> Great start @anishaswain! A few design considerations & bugs we'll need to address before merging:
> 
> * Hovering over the toggled filter selection component highlights every field name in red
> * The width of the dropdown menu for a field scales according to the page width. Can we consider setting `width: auto` to set the width for the menu?
> * Clicking the "Save" button doesn't close the toggled filter selection component
> * We can utilize the [Toolbar](https://www.patternfly.org/v4/components/toolbar) component to align the "Apply Filters" and "Clear Filters" buttons with "Compare Iterations"

FIXED the above points and added `ToolBar` component

> * Closing a selected filter from a field's dropdown menu doesn't remove the selected filter

@gurbirkalsi this feature is not functional in the `main` branch itself. I haved opened up another PR #164 which will fix this bug

aquibbaig commented:
@anishaswain can you check why the dropdown floats to the left in my case?
<img width="1439" alt="Screenshot" src="https://user-images.githubusercontent.com/26324376/124456104-f3883900-dda7-11eb-9ac3-8b0d85e95bed.png">

anishaswain commented:
> @anishaswain can you check why the dropdown floats to the left in my case?

@aquibbaig could you please help me with your screen resolution? I am not being able to reproduce the issue.

It is a Mac just like yours, 2560 x 1600 probably

aquibbaig commented:
this issue has now resolved @anishaswain, it looks great

aquibbaig commented on src/components/TableFilterSelection/index.js:
```diff
@@ -51,49 +57,66 @@ export default class TableFilterSelection extends Component {
+    ];
+
+    return (
+      <div id="dropdownDiv">
```
are we using this id somewhere?

anishaswain replied:
fixed!

dbutenhof commented on src/components/TableFilterSelection/index.js:
```diff
@@ -51,49 +57,66 @@ export default class TableFilterSelection extends Component {
+            >
+              Save
+            </Button>
+            {/* </div> */}
```
Is the commented `</div>` here inside empty brackets important somehow?

anishaswain replied:
Node, left it by by mistake. Resolved now.

dbutenhof reviewed (approved):
So far as I can tell, this looks OK. Compared to the early screen shots, combining the filter and compare buttons into a toolbar looks like an improvement.

One minor comment/question ...

aquibbaig commented:
@anishaswain any updates on this PR? I think it's ready for a merge

anishaswain commented:
> @anishaswain any updates on this PR? I think it's ready for a merge

@aquibbaig yes, we should merge it.
